### DISCUSSION TITLE
Are there any plans to support llvm builds without RTTI?

### DISCUSSION THREAD

**Discussion by nsmithtt** [onnx/onnx-mlir#2897] [Q&A]
Having it enabled is unconventional and makes this project less portable.

**Comment by AlexandreEichenberger** [comment]
@gongsu832 can you chime in?

**Comment by gongsu832** [comment]
I have no idea what RTTI is.

**Reply by nsmithtt** [reply]
Runtime type information, it's enabled here: https://github.com/onnx/onnx-mlir/blob/main/utils/build-mlir.sh#L9

It allows you to use things like `dynamic_cast`.  LLVM has their own RTTI system though that is much more performant, and typically LLVM distributions do not have RTTI enabled.

**Reply by nsmithtt** [reply]
Currently ONNX-MLIR build fails without an RTTI enabled llvm
